OpenAI/Packages/com.openai.unity/Runtime/Threads/CreateRunRequest.cs

```diff
@@ -71,6 +237,80 @@ public CreateRunRequest(string assistantId, string model = null, string instruct
+        /// <summary>
+        /// An alternative to sampling with temperature, called nucleus sampling,
+        /// where the model considers the results of the tokens with top_p probability mass.
+        /// So 0.1 means only the tokens comprising the top 10% probability mass are considered.
+        /// We generally recommend altering this or temperature but not both.
+        /// </summary>
+        [Preserve]
+        [JsonProperty("top_p")]
+        public double? TopP { get; }
+
+        /// <summary>
+        /// If true, returns a stream of events that happen during the Run as server-sent events,
+        /// terminating when the Run enters a terminal state with a 'data: [DONE]' message.
+        /// </summary>
+        [Preserve]
+        [JsonProperty("stream")]
+        public bool Stream { get; internal set; }
+
+        /// <summary>
+        /// The maximum number of prompt tokens that may be used over the course of the run.
+        /// The run will make a best effort to use only the number of prompt tokens specified,
+        /// across multiple turns of the run. If the run exceeds the number of prompt tokens specified,
+        /// the run will end with status 'incomplete'. See 'incomplete_details' for more info.
+        /// </summary>
+        [Preserve]
+        [JsonProperty("max_prompt_tokens")]
+        public int? MaxPromptTokens { get; }
+
+        /// <summary>
+        /// The maximum number of completion tokens that may be used over the course of the run.
+        /// The run will make a best effort to use only the number of completion tokens specified,
+        /// across multiple turns of the run. If the run exceeds the number of completion tokens specified,
+        /// the run will end with status 'incomplete'. See 'incomplete_details' for more info.
+        /// </summary>
+        [Preserve]
+        [JsonProperty("max_completion_tokens")]
+        public int? MaxCompletionTokens { get; }
+
+        /// <summary>
+        /// Controls for how a thread will be truncated prior to the run.
+        /// Use this to control the initial context window of the run.
+        /// </summary>
+        [Preserve]
+        [JsonProperty("truncation_strategy")]
+        public TruncationStrategy TruncationStrategy { get; }
+
+        /// <summary>
+        /// Controls which (if any) tool is called by the model.
+        /// none means the model will not call any tools and instead generates a message.
+        /// auto is the default value and means the model can pick between generating a message or calling one or more tools.
+        /// required means the model must call one or more tools before responding to the user.
+        /// Specifying a particular tool like {"type": "file_search"} or {"type": "function", "function": {"name": "my_function"}}
+        /// forces the model to call that tool.
+        /// </summary>
+        [Preserve]
+        [JsonProperty("tool_choice")]
+        public dynamic ToolChoice { get; }
```

Comment:
@StephenHodgson I believe the `dynamic` keyword isn't supported on all platforms like WebGL and requires .NET 4.0 (doesn't work w/ 2.1).

Any chance we can avoid using `dynamic`?

Reply:
It's being discussed a bit here too with Unity stating it's not well supported:
https://forum.unity.com/threads/missing-compiler-required-member-microsoft-csharp-runtimebinder-csharpargumentinfo-create.563839/#post-8865384

Reply:
Oh right, I meant to convert those back to `object`
```suggestion
        public object ToolChoice { get; }
```

Reply:
Wow. Lightning fast :) That's great! Thanks so much.